animations/hero-animations/index.md

```diff
@@ -0,0 +1,670 @@
+
+* Use two hero widgets with matching tags to implement the transition.
+* The Navigator manages a stack containing the app’s routes.
+* Pushing a route on or popping a route from the Navigator’s stack
+  triggers the transition.
+* The Flutter framework calculates a rectangle tween that defines
+  the hero's boundary as it flies from the source to the destination
+  route. During its flight, the hero is moved to an application
+  overlay, so that it appears on top of both routes.
+</div>
+
+<aside class="alert alert-info" markdown="1">
+**Terminology:**
+If the concept of tweens or tweening is new to you, please see the
+[Animations in Flutter tutorial.](/tutorials/animation/)
+</aside>
+
+Hero animations are implemented using two
+[Hero](https://docs.flutter.io/flutter/widgets/Hero-class.html)
+widgets: one describing the widget in the source route,
+and another describing the widget in the destination route.
+From the user’s point of view, the hero appears to be shared, and
+only the programmer needs to understand this implementation detail.
+
+<aside class="alert alert-info" markdown="1">
+**Note about dialogs:**
+Heroes fly from one PageRoute to another. Dialogs
+(displayed with `showDialog()`, for example), use PopupRoutes,
+which are not PageRoutes.  At least for now,
+you can't animate a hero to a Dialog.
+For further developments (and a possible workaround), [watch this
+issue.](https://github.com/flutter/flutter/issues/10667)
+</aside>
+
+Hero animation code has the following structure:
+
+1. Define a starting Hero object, referred to as the _source
+   hero_. The hero specifies its location,
+   size, graphical representation (typically an image), and an
+   identifying tag.
+1. Define an ending Hero object, referred to as the _destination hero_.
+   This hero also specifies its location, size, graphical
+   representation, and the same tag as the source hero.
+   It's <strong>essential that both hero objects are created with
+   the same tag</strong>, typically an object that represents the
+   underlying data. For best results, the heroes should have
+   virtually identical widget trees.
+1. Create a route that contains the destination hero.
+   The destination route defines the widget tree that exists
+   at the end of the animation.
+1. Trigger the animation by pushing the route on the Navigator's stack.
+   The Navigator push and pop operations trigger a hero animation
+   for each pair of heroes with matching tags in the source
+   and destination routes.
+
+Flutter calculates the tweens required to transition the
+starting hero to the ending hero (interpolating size and
+position changes), and performs the animation in an overlay.
+
+The next section describes Flutter's process in greater detail.
+
+## Behind the scenes
+
+The following describes how Flutter performs the transition.
```

Comment:
"transition" may not be clear to the reader, at least not yet. Might be helpful to explain that you're talking about the transition from one route to the other...